tfhe/src/core_crypto/commons/math/ntt/ntt64.rs

```diff
@@ -0,0 +1,121 @@
+use crate::core_crypto::commons::ciphertext_modulus::CiphertextModulusKind;
+use crate::core_crypto::commons::utils::izip;
+use crate::core_crypto::prelude::*;
+use concrete_ntt::prime64::Plan;
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+use std::sync::{Arc, OnceLock, RwLock};
+
+#[derive(Clone, Debug)]
+pub struct Ntt64 {
+    plan: Arc<Plan>,
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct Ntt64View<'a> {
+    pub(crate) plan: &'a Plan,
+}
+
+impl Ntt64 {
+    #[inline]
+    pub fn as_view(&self) -> Ntt64View<'_> {
+        Ntt64View { plan: &self.plan }
+    }
+}
+
+type PlanMap = RwLock<HashMap<usize, Arc<OnceLock<Arc<Plan>>>>>;
```

Comment:
Could be replaced by a concurrent hashmap like https://github.com/xacrimon/dashmap

Reply:
Also, is the `OnceLock` needed here?
It seems to me that we would insert already built `Plan`s in the HashMap

Reply:
So this was what was done initially by Sarah, I have to say I did not recheck the details, we have the same thing for the fft, don't know about the concurrent hashmap as it does not seem like we *really* need it and we try to an extent to limit the dependencies we pull in

Reply:
the second oncelock is to avoid a race condition where two threads may have gotten a copy of the Arc<Plan> and make sure it is only initialized once